core/controller/src/main/scala/org/apache/openwhisk/core/controller/Controller.scala

```diff
@@ -151,6 +153,16 @@ class Controller(val instance: ControllerInstanceId,
             .invokerHealth()
             .map(_.count(_.status == InvokerState.Healthy).toJson)
         }
+      } ~ path("ready") {
+        onSuccess(loadBalancer.invokerHealth()) { invokersHealth =>
```

Comment:
minor style suggestion (take it or leave it): since non-success is a 500, you might be able to simplify this to:
```
        complete {
          loadBalancer
            .invokerHealth()
            .map(
              invokersHealth =>
                Controller.readyState(
                  invokersHealth.size,
                  invokersHealth.count(_.status == InvokerState.Healthy),
                  Controller.readinessThreshold.getOrElse(1)))
        }
```
Then, make Controller.readinessThreshold return a JsObject or throw an exception with the desired error message.

Reply:
I was thinking of doing that originally but that implied writing an Exception Handler and though it would not make a lot of sense of doing that for only one endpoint